artiq/coredevice/phaser.py

```diff
@@ -1073,9 +1073,7 @@ def set_servo(self, profile=0, bypass=1, hold=0):
             data = 1
         if hold == 1:
             data = data | (1 << 1)
-        if bypass:
-            hold = 1
-        data = (profile << 2) | (hold << 1) | (bypass << 0)
```

Comment:
Why don't you just do this? Take a step back and look at it.

Reply:
My thinking was that then an invalid value for hold or bypass would not affect the other parts of the register. Should I check that hold and bypass are valid (0 or 1)?

Reply:
Ok. But then just `& 1` them in the expression.